src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptAsScriptingOperation.cs

```diff
@@ -0,0 +1,322 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using Microsoft.SqlTools.ServiceLayer.Scripting.Contracts;
+using Microsoft.SqlTools.Utility;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Smo;
+using System.Collections.Specialized;
+using System.Text;
+using System.Globalization;
+using Microsoft.SqlServer.Management.SqlScriptPublish;
+
+namespace Microsoft.SqlTools.ServiceLayer.Scripting
+{
+    /// <summary>
+    /// Class to generate script as for one smo object
+    /// </summary>
+    public class ScriptAsScriptingOperation : SmoScriptingOperation
+    {
+        public ScriptAsScriptingOperation(ScriptingParams parameters): base(parameters)
+        {
+        }
+
+        private string BatchTerminator = "GO";
+
+        public override void Execute()
+        {
+            SqlServer.Management.Smo.Scripter scripter = null;
+            try
+            {
+                this.CancellationToken.ThrowIfCancellationRequested();
+
+                this.ValidateScriptDatabaseParams();
+
+                this.CancellationToken.ThrowIfCancellationRequested();
+                string resultScript = string.Empty;
+                // TODO: try to use one of the existing connections
+                using (SqlConnection sqlConnection = new SqlConnection(this.Parameters.ConnectionString))
+                {
+                    sqlConnection.Open();
+                    ServerConnection serverConnection = new ServerConnection(sqlConnection);
+                    Server server = new Server(serverConnection);
+                    scripter = new SqlServer.Management.Smo.Scripter(server);
+                    ScriptingOptions options = new ScriptingOptions();
+                    SetScriptBehavior(options);
+                    PopulateAdvancedScriptOptions(this.Parameters.ScriptOptions, options);
+                    options.WithDependencies = false;
```

Comment:
I wonder should scripting dependencies be a config option too? Not vital, just something to think about